File: amic-platform/src/modules/ma/types/dd_checklist.ts
export type DDWorkstream =
  // FDD (재무실사)
  | "FDD_FINANCIAL_STATEMENTS"
  | "FDD_REVENUE"
  | "FDD_WORKING_CAPITAL"
  | "FDD_DEBT_CASH"
  | "FDD_PROJECTIONS"
  // LDD (법률실사)
  | "LDD_CORPORATE"
  | "LDD_PERMITS"
  | "LDD_CONTRACTS"
  | "LDD_ASSETS"
  | "LDD_LABOR"
  | "LDD_LITIGATION"
  | "LDD_IP"
  | "LDD_INSURANCE"
  | "LDD_ENVIRONMENT"
  // TDD (세무실사)
  | "TDD_CORPORATE_TAX"
  | "TDD_VAT"
  | "TDD_TRANSFER_PRICING"
  | "TDD_WITHHOLDING"
  | "TDD_TAX_INCENTIVES"
  // 기타
  | "OTHER";

export type DDChecklistStatus =
  | "NOT_STARTED"
  | "IN_PROGRESS"
  | "COMPLETED"
  | "NOT_APPLICABLE";

export interface DDChecklistItem {
  id: string;
  transaction_id: string;
  workstream: DDWorkstream;
  title: string;
  description: string | null;
  assignee_email: string | null;
  status: DDChecklistStatus;
  due_date: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface DDChecklistCreate {
  workstream: DDWorkstream;
  title: string;
  description?: string;
  assignee_email?: string;
  due_date?: string;
  notes?: string;
}

export interface DDChecklistUpdate {
  workstream?: DDWorkstream;
  title?: string;
  description?: string;
  assignee_email?: string;
  status?: DDChecklistStatus;
  due_date?: string;
  notes?: string;
}

export interface DDWorkstreamSummary {
  workstream: DDWorkstream;
  total: number;
  completed: number;
  in_progress: number;
  not_started: number;
}

export interface DDChecklistSummary {
  total: number;
  by_workstream: DDWorkstreamSummary[];
  overall_completion_pct: number;
}

File: deal-mgmt/app/schemas/dd_checklist.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import DDChecklistStatus, DDWorkstream


class DDChecklistOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    transaction_id: uuid.UUID
    workstream: DDWorkstream
    title: str
    description: str | None = None
    assignee_email: str | None = None
    status: DDChecklistStatus
    due_date: str | None = None
    notes: str | None = None
    created_at: datetime
    updated_at: datetime


class DDChecklistCreate(BaseModel):
    workstream: DDWorkstream
    title: str = Field(..., min_length=1, max_length=300)
    description: str | None = None
    assignee_email: str | None = Field(None, max_length=255)
    due_date: str | None = Field(None, max_length=10)
    notes: str | None = None


class DDChecklistUpdate(BaseModel):
    workstream: DDWorkstream | None = None
    title: str | None = Field(None, min_length=1, max_length=300)
    description: str | None = None
    assignee_email: str | None = Field(None, max_length=255)
    status: DDChecklistStatus | None = None
    due_date: str | None = Field(None, max_length=10)
    notes: str | None = None


class DDWorkstreamSummary(BaseModel):
    workstream: DDWorkstream
    total: int
    completed: int
    in_progress: int
    not_started: int


class DDChecklistSummary(BaseModel):
    total: int
    by_workstream: list[DDWorkstreamSummary]
    overall_completion_pct: float
